client/src/notebooks/Notebooks.container.js

```diff
@@ -451,7 +451,7 @@ class StartNotebookServer extends Component {
   setServerOptionFromEvent(option, event, providedValue = null) {
     const target = event.target.type.toLowerCase();
     let value = providedValue;
-    if (!providedValue) {
+    if (!providedValue && providedValue !== 0) {
```

Comment:
```suggestion
    if (!providedValue != null) {
```

I think the logic here is clearer if we test for `null`, especially if we consider, in addition to `providedValue === 0` the possibility of what happens when `providedValue === false`.